stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -138,8 +157,11 @@ public void onHalfClose() {
                 onReady();
               }
             } else {
-              call.close(Status.INTERNAL.withDescription("Half-closed without a request"),
-                  new Metadata());
+              Status internalError = Status.INTERNAL.withDescription(MISSING_REQUEST);
+              StatusException exception = internalError.asException();
+              log.log(Level.FINE, internalError.getDescription(), exception);
```

Comment:
It seems fine to me not to log this, since the client will be told pretty clearly what went wrong and the backtrace in the exception probably isn't too useful.

However, if you are going to log it, don't create the exception unless logging is on. So surround the two lines here with  a `logger.isLoggable(FINE)` check. Ditto elsewhere.